src/net/sourceforge/kolmafia/textui/langserver/Script.java

```diff
@@ -0,0 +1,193 @@
+package net.sourceforge.kolmafia.textui.langserver;
+
+import java.io.File;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import net.sourceforge.kolmafia.textui.Parser;
+import net.sourceforge.kolmafia.textui.Parser.AshDiagnostic;
+import net.sourceforge.kolmafia.textui.parsetree.Scope;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.PublishDiagnosticsParams;
+
+/**
+ * A file that was recognized as a KoLmafia script in one of the directories under our authority.
+ */
+public class Script {
+  final AshLanguageServer parent;
+  final File file;
+
+  /**
+   * The {@link Handler} in charge of this script. <i>Only</i> exists if this script is <b>not</b>
+   * imported by any other script.
+   *
+   * <p>When trying to find the handler(s) in charge of a given file, {@link
+   * FilesMonitor#findHandlers(File)} should be used.
+   */
+  Handler handler;
+
+  Script(final AshLanguageServer parent, final File file) {
+    this.parent = parent;
+    this.file = file;
+  }
+
+  Handler makeHandler() {
+    this.handler = new Handler();
+
+    this.parent.executor.execute(
+        () -> {
+          this.handler.parseFile();
+        });
+
+    return this.handler;
+  }
```

Comment:
Unless you have a good reason to, I'd prefer to avoid the default visibility modifier in favour of one of public, protected and private.

Good reasons I'm aware of include:
* wanting a field / method to be accessible in a test (accompanied with a comment saying it should be private)
* splitting a series of former nested classes into their own namespace, and not wanting them to be generally public

Reply:
This also applies to everywhere else in this commit you've used default visibility modifiers. For example, in `AshLanguageServerTest` I'd expect most things to go to `protected`, because the class is designed to be inherited.

Reply:
I completely agree with wanting explicit private, protected, or public for methods and fields.

Reply:
these are all on purpose, though.

These are purposefully package-only visibility.
There's no point in them being protected since this is not a class meant to be extended.

Reply:
You're not listening. Use explicit keywords, rather than omitting them.

Reply:
:-(

Reply:
I appreciate that was probably more of a response to me, giving your reasons, but I think Veracity would prefer we forbid the default modifiers entirely in this case. I think, stylistically, I agree: `protected` should serve the same purpose (of forbidding use outside the package) if the class is never extended, and reads more easily to me (I, being somewhat old-school, don't see many things use the package local modifier). It also means that if someone wanted to extend the class for some reason, everything would work as hoped for.

Reply:
> I agree: `protected` should serve the same purpose (of forbidding use outside the package) if the class is never extended, and reads more easily to me. It also means that if someone wanted to extend the class for some reason, everything would work as hoped for.

The only issue I have with this is that seeing "protected" *implies* that the class *can* be extended. So if the class can't or is not meant to, it creates some confusion IMO.

Reply:
I acknowledge that that could be an issue, but I'd still like you to avoid using the default modifiers, as Veracity requested.

Reply:
If the intent is that the class cannot be extended, why is it not final?

Reply:
The point was that even if you were to try, you wouldn't be able to replace the one the server is using, making it effectively pointless to extend, but I guess also marking it as final can't hurt.